src/hooks/useFileUpload.ts

```diff
@@ -0,0 +1,183 @@
+        const uploadResponse = await authenticatedFetchWithExpiration(
+          API_CONFIG.buildUrl(
+            API_CONFIG.ENDPOINTS.UPLOAD.DIRECT(tenant, filename)
+          ),
+          {
+            method: "PUT",
+            jwt,
+            body: file,
+            headers: {
+              "Content-Type": file.type || "application/pdf",
+            },
+          }
+        );
+
+        if (uploadResponse.jwtExpired) {
+          throw new Error("Authentication expired. Please log in again.");
+        }
+
+        if (!uploadResponse.response.ok) {
+          const errorText = await uploadResponse.response.text();
+          throw new Error(
+            `Upload failed: ${uploadResponse.response.status} ${errorText}`
+          );
+        }
+
+        // Emit upload completed event
+        emit({
+          type: EVENT_TYPES.FILE_UPLOAD.COMPLETED,
+          fileKey,
+          filename,
+          progress: 100,
+          source: "useFileUpload",
+        } as FileUploadEvent);
+
+        // Success state - trigger AutoRAG sync and start polling
+        setUploadedFileInfo({
+          filename: filename,
+          fileKey: fileKey,
+        });
+
+        // Send notification about successful upload
+        if (onSendNotification) {
+          onSendNotification(
+            `Your recent document "${filename}" has been indexed and is ready to be added to a campaign.`
+          );
+        }
+
+        // Call success callback
+        onUploadSuccess?.(filename, fileKey);
+
+        // Trigger AutoRAG sync and start polling for job status
+        try {
+          // Add a small delay to avoid hitting rate limits
+          await new Promise((resolve) => setTimeout(resolve, 1000));
+
+          const ragId = AUTORAG_CONFIG.LIBRARY_RAG_ID;
+          const jobId = await AutoRAGService.triggerSync(ragId);
+
+          console.log("[useFileUpload] AutoRAG sync triggered, job_id:", jobId);
+
+          // Start polling for job status with file key
+          startPolling(ragId, jobId, fileKey);
+
+          // Add a cooldown period to prevent hitting rate limits
+          setTimeout(() => {
+            // 5 second cooldown to prevent hitting rate limits
+          }, 5000);
+        } catch (syncError) {
+          console.error("[useFileUpload] AutoRAG sync error:", syncError);
+        }
+      } catch (error) {
+        console.error("[useFileUpload] Upload error:", error);
+
+        // Emit upload failed event
+        emit({
+          type: EVENT_TYPES.FILE_UPLOAD.FAILED,
+          fileKey: buildAutoRAGFileKey(
+            AuthService.getUsernameFromStoredJwt() || "",
+            filename
+          ),
+          filename,
+          error: error instanceof Error ? error.message : "Unknown error",
+          source: "useFileUpload",
+        } as FileUploadEvent);
+      }
+    },
+    [emit, startPolling, onSendNotification, onUploadSuccess]
+  );
+
+  const clearUploadedFileInfo = useCallback(() => {
```

Comment:
when is this called? do we care to upload multiple files in the same campaign now? (would support multi-select in the ui)